addons/web/static/src/search/search_bar/search_bar.xml

```diff
@@ -107,7 +107,7 @@
                         role="img"
                     />
                 </button>
-                <div class="o_searchview_input_container d-flex flex-grow-1 flex-wrap gap-1">
+                <div class="o_searchview_input_container d-flex flex-grow-1 flex-wrap gap-1 text-truncate">
```

Comment:
Not sure which effect you tried to accomplish, but it doesn't truncate with an ellipsis...